services/api-server/src/simcore_service_api_server/core/_prometheus_instrumentation.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+from dataclasses import dataclass
+from typing import Final, cast
+
+from attr import field
+from fastapi import FastAPI
+from prometheus_client import CollectorRegistry, Gauge
+from pydantic import PositiveInt
+from servicelib.fastapi.prometheus_instrumentation import (
+    setup_prometheus_instrumentation as setup_rest_instrumentation,
+)
+from simcore_service_api_server.api.dependencies.rabbitmq import (
+    get_log_distributor,
+    wait_till_log_distributor_ready,
+)
+from simcore_service_api_server.models.schemas.jobs import JobID
+
+from .._meta import PROJECT_NAME
+
+METRICS_NAMESPACE: Final[str] = PROJECT_NAME.replace("-", "_")
+
+
+@dataclass(slots=True, kw_only=True)
+class ApiServerPrometheusInstrumentation:
+    registry: CollectorRegistry
+    _logstreaming_queues: Gauge = field(init=False)
+
+    def __post_init__(self) -> None:
+        self._logstreaming_queues = Gauge(
+            "log_stream_queue_length",
+            "#Logs in log streaming queue",
+            ["job_id"],
+            namespace=METRICS_NAMESPACE,
+        )
+
+    def update_metrics(self, log_queue_sizes: dict[JobID, int]):
+        self._logstreaming_queues.clear()
+        for job_id, length in log_queue_sizes.items():
+            self._logstreaming_queues.labels(job_id=job_id).set(length)
+
+
+async def collect_prometheus_metrics_task(app: FastAPI):
+    metrics_collect_seconds: PositiveInt = (
+        app.state.settings.API_SERVER_PROMETHEUS_INSTRUMENTATION_COLLECT_SECONDS
+    )
+    assert (
+        app.state.instrumentation
+    ), "Instrumentation not setup. Please check the configuration"  # nosec
+    instrumentation = get_instrumentation(app)
+    await wait_till_log_distributor_ready(app)
+    log_distributor = get_log_distributor(app)
+    while True:
+        await asyncio.sleep(metrics_collect_seconds)
+        instrumentation.update_metrics(
```

Comment:
actually we do have something called a background task that you could use instead of this:
https://github.com/ITISFoundation/osparc-simcore/blob/194c626059600ff1a2cec44ff43d45a5eda93c86/packages/service-library/src/servicelib/background_task.py

Reply:
Thanks a lot for the tip. I changed this to our background task